### PR TITLE
Documentation for Observation R4 charted search

### PR DESCRIPTION
**Pull request by BS062721**

### CODE REVIEW

**Comment by BS062721** [comment]
auth_type
Diagnostic_menu
Extensions
grouper_code_search
Observation_example
Proprietary_code_diagnostic
Proprietary_code_observation
Proprietary_code_set_52
subfeilds

**Comment by SA030800** [comment]
@MadhurRaj , @MaxPhilips  would you help us in reviewing this PR

**Comment by BS062721** [comment]
Updated screenshots:
Screen Shot 2020-02-13 at 10 01 03 AM
Screen Shot 2020-02-13 at 10 01 15 AM
Screen Shot 2020-02-13 at 10 00 43 AM
Screen Shot 2020-02-13 at 10 01 21 AM

File: content/millennium/r4/diagnostic/observation.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+* TOC
+{:toc}
+
+## Overview
+The Observation resource provides measurements or simple assertions about a patient that are useful for establishing baselines or trends, monitoring a patient's progress, and establishing diagnoses. Most observations are simple name/value pair assertions but some observations, such as blood pressure, group other observations together logically. Examples of common observations are: Laboratory results (blood sugar, hemoglobin), Vital signs (temperature, blood pressure), Personal characteristics (height, weight), and Social history (tobacco/alcohol use, employment status). Pathology reports, radiology reports, and other textual reports should be represented by the DiagnosticReport resource.
+
+The following fields are returned if valued:
+
+* [Id](http://hl7.org/fhir/R4/resource-definitions.html#Resource.id){:target="_blank"}
```
Comment (MadhurRaj) [inline content/millennium/r4/diagnostic/observation.md]: all these links need to be https://

File: content/millennium/r4/diagnostic/observation.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+#### Response
+
+<%= headers status: 200 %>
+<%= json(:R4_OBSERVATION_BUNDLE) %>
+
+### Errors
+
+The common [errors] and [OperationOutcomes] may be returned.
+
+[`reference`]: http://hl7.org/fhir/r4/search.html#reference
```
Comment (MadhurRaj) [inline content/millennium/r4/diagnostic/observation.md]: these have to be https:// too.

File: layouts/devices_r4.html [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<h1 id="devices">Devices</h1>
+<p>The Device resources are used to manage and track devices. Devices can be many things including implants, medical
+  equipment, and software, and they can produce other resources (such as Observations).</p>
+
+<p>Click <a href="http://hl7.org/fhir/r4/administration-module.html#dev-sub">here</a> to see the detailed categorization and full set of
```
Comment (MadhurRaj) [inline layouts/devices_r4.html]: https here too

File: layouts/diagnostic_r4.html [hunk 4]
```diff
@@ -0,0 +1,13 @@
+<h1 id="diagnostic">Diagnostic</h1>
+
+<p>The Diagnostic resources are concerned with the process of observing, testing, and diagnosing a patient. Note, once a
+  diagnosis is confirmed, it would be recorded in Condition so that it is part of the historical patient record.</p>
+
+<p>The Observation resource is primarily concerned with recording discrete values, answers, and results such as clinical
+  laboratory data and vital signs while the DiagnosticReport resource typically provides a textual set of information and
+  interpretation after performing a diagnostic service such as a Radiology report or Pathology report.</p>
+
+<p>Click <a href="http://hl7.org/fhir/r4/diagnostics-module.html">here</a> to see the detailed categorization and full set of
```
Comment (MadhurRaj) [inline layouts/diagnostic_r4.html]: same here.

File: lib/resources/r4/observation.yaml [hunk 5]
```diff
@@ -0,0 +1,116 @@
+---
+name: Observation
+field_name_base_url: http://hl7.org/fhir/R4/observation-definitions.html#Observation
```
Comment (MadhurRaj) [inline lib/resources/r4/observation.yaml]: https

File: lib/resources/r4/observation.yaml [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+  required: 'No'
+  cardinality: 0..*
+  type: CodeableConcept
+  description: A code that classifies the general type of observation being made.
+  binding:
+    description: Codes for high level observation categories.
+    terminology:
+    - display: Observation Category Codes
+      system: http://terminology.hl7.org/CodeSystem/observation-category
+      info_link: http://hl7.org/fhir/R4/valueset-observation-category.html
```
Comment (MadhurRaj) [inline lib/resources/r4/observation.yaml]: info_links should be https

File: lib/resources/r4/observation.yaml [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+  required: 'No'
+  cardinality: 0..*
+  type: CodeableConcept
+  description: A categorical assessment of an observation value. For example, high, low, normal.
+  binding:
+    description: Codes identifying interpretations of observations.
+    terminology:
+    - display: v3 Interpretation Codes
+      system: http://terminology.hl7.org/CodeSystem/v3-ObservationInterpretation
+      info_link: http://terminology.hl7.org/CodeSystem/v3-ObservationInterpretation
```
Comment (MadhurRaj) [inline lib/resources/r4/observation.yaml]: this is not an info link.

File: lib/resources/r4/observation.yaml [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+    required: 'No'
+    cardinality: 0..*
+    type: CodeableConcept
+    description: A categorical assessment of an observation value. For example, high, low, normal.
+    binding:
+      description: Codes identifying interpretations of observations.
+      terminology:
+      - display: v3 Interpretation Codes
+        system: http://terminology.hl7.org/CodeSystem/v3-ObservationInterpretation
+        info_link: http://terminology.hl7.org/CodeSystem/v3-ObservationInterpretation
```
Comment (MadhurRaj) [inline lib/resources/r4/observation.yaml]: same here.

File: lib/resources/example_json/r4_examples_observation.rb [hunk 9]
```diff
@@ -0,0 +1,2458 @@
+module Cerner
+    module Resources
+  
+        R4_OBSERVATION_BUNDLE ||=
+        {
```
Comment (MadhurRaj) [inline lib/resources/example_json/r4_examples_observation.rb]: this example is way too large to display.

File: lib/resources/r4/observation.yaml (hunk 7 above)
Comment (SA030800) [reply lib/resources/r4/observation.yaml]: actually looks like the info link should be: https://hl7.org/fhir/r4/valueset-observation-interpretation.html

File: lib/resources/r4/observation.yaml (hunk 8 above)
Comment (SA030800) [reply lib/resources/r4/observation.yaml]: https://hl7.org/fhir/r4/valueset-observation-interpretation.html

File: content/millennium/r4/diagnostic/observation.md (hunk 1 above)
Comment (BS062721) [reply content/millennium/r4/diagnostic/observation.md]: https://github.com/cerner/fhir.cerner.com/pull/345/commits/85bb1a56bd4635bc292f0fbe1300685b9970981e#diff-d42a2575832583fdd3d20dff4a54f294R15

File: content/millennium/r4/diagnostic/observation.md (hunk 2 above)
Comment (BS062721) [reply content/millennium/r4/diagnostic/observation.md]: https://github.com/cerner/fhir.cerner.com/pull/345/commits/85bb1a56bd4635bc292f0fbe1300685b9970981e#diff-d42a2575832583fdd3d20dff4a54f294R114

File: layouts/devices_r4.html (hunk 3 above)
Comment (BS062721) [reply layouts/devices_r4.html]: https://github.com/cerner/fhir.cerner.com/pull/345/commits/85bb1a56bd4635bc292f0fbe1300685b9970981e#diff-9a20c687785d0a97b0cf7c8d35ff645dR10

File: layouts/diagnostic_r4.html (hunk 4 above)
Comment (BS062721) [reply layouts/diagnostic_r4.html]: https://github.com/cerner/fhir.cerner.com/pull/345/commits/85bb1a56bd4635bc292f0fbe1300685b9970981e#diff-9a20c687785d0a97b0cf7c8d35ff645dR10

File: lib/resources/r4/observation.yaml (hunk 6 above)
Comment (BS062721) [reply lib/resources/r4/observation.yaml]: https://github.com/cerner/fhir.cerner.com/pull/345/commits/85bb1a56bd4635bc292f0fbe1300685b9970981e#diff-53f5b7c6241dd914461705ddc9523c7eR3

File: lib/resources/r4/observation.yaml (hunk 7 above)
Comment (BS062721) [reply lib/resources/r4/observation.yaml]: https://github.com/cerner/fhir.cerner.com/pull/345/commits/85bb1a56bd4635bc292f0fbe1300685b9970981e#diff-53f5b7c6241dd914461705ddc9523c7eR58

File: lib/resources/r4/observation.yaml (hunk 8 above)
Comment (BS062721) [reply lib/resources/r4/observation.yaml]: https://github.com/cerner/fhir.cerner.com/pull/345/commits/85bb1a56bd4635bc292f0fbe1300685b9970981e#diff-53f5b7c6241dd914461705ddc9523c7eR101

File: lib/resources/example_json/r4_examples_observation.rb (hunk 9 above)
Comment (BS062721) [reply lib/resources/example_json/r4_examples_observation.rb]: trimmed it down: https://github.com/cerner/fhir.cerner.com/pull/345/commits/85bb1a56bd4635bc292f0fbe1300685b9970981e#diff-cd059d38e44f892deed0ab35bef7831cL501

File: lib/resources/r4/observation.yaml (hunk 5 above)
Comment (BS062721) [reply lib/resources/r4/observation.yaml]: https://github.com/cerner/fhir.cerner.com/pull/345/commits/85bb1a56bd4635bc292f0fbe1300685b9970981e#diff-53f5b7c6241dd914461705ddc9523c7eR3

File: content/millennium/r4/diagnostic/observation.md [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+* When multiple pages of Observation results are returned for a single query:
+  * All Social history Observations (if any qualify for the query) will be returned on the first page of results. This means that the next bullet does not apply to Social history Observations.
+  * Results are sorted by effective date/time in descending order by page. That is, all Observations on any given page of results are newer than all Observations on the next page of results. Sort order within pages is not guaranteed.
+  * If the query uses the `_lastUpdated` query parameter, results are sorted by last updated date/time in descending order by page, not by effective date/time.
+
+### Parameters
+
+ Name             | Required?         | Type          | Description
+------------------|-------------------|---------------|-----------------------------------------------------------------------------------------------------------------------------------------
+ `patient`        | This or `subject` | [`reference`] | The subject that the observation is about (if patient). Example: `patient=1316024`
```
Comment (MadhurRaj) [inline content/millennium/r4/diagnostic/observation.md]: can't we say  'The Patient that the observation is about'?

File: content/millennium/r4/diagnostic/observation.md [hunk 11]
```diff
@@ -0,0 +1,122 @@
...
+ `subject`        | This or `patient` | [`reference`] | The subject (Patient) that the observation is about. Example: `subject=Patient/1316024` or `subject:Patient=1316024`
+ `code`           | N                 | [`token`]     | The code or component-code of the observation type. Example: `code=http://loinc.org|3094-0,http://loinc.org|3139-3`
+ `date`           | N                 | [`date`]      | Date range into which the observation falls. Example: `date=gt2014-09-24` or `date=lt2015-09-24T12:00:00.000Z`
+ `_lastUpdated`   | N                 | [`date`]      | Date range in which the observation was last updated. Example: `_lastUpdated=gt2014-09-24` or `_lastUpdated=lt2015-09-24T12:00:00.000Z`
+ `category`       | N                 | [`token`]     | The category of observations. Example: `category=laboratory`
+ [`_count`]       | N                 | [`number`]    | The maximum number of results to return per page. Defaults to `50`.
```
Comment (MadhurRaj) [inline content/millennium/r4/diagnostic/observation.md]: Remove the default value. We don't need to advertise that.

File: content/millennium/r4/diagnostic/observation.md [hunk 12]
```diff
@@ -0,0 +1,122 @@
...
+
+#### Request
+
+    GET https://fhir-open.sandboxcerner.com/r4/0b8a0111-e8e6-4c26-a91c-5069cbc6b1ca/Observation?patient=3998008
+
+#### Response
+
+<%= headers status: 200 %>
+<%= json(:R4_OBSERVATION_BUNDLE) %>
+
```
Comment (MadhurRaj) [inline content/millennium/r4/diagnostic/observation.md]: let's add the disclaimer here.
https://github.com/cerner/fhir.cerner.com/blame/master/content/millennium/r4/financial/coverage.md#L79

File: content/millennium/r4/diagnostic/observation.md (hunk 11 above)
Comment (BS062721) [reply content/millennium/r4/diagnostic/observation.md]: https://github.com/cerner/fhir.cerner.com/pull/345/commits/3756e931418b52a36c0e6998015daa0d7adbb1c5#diff-d42a2575832583fdd3d20dff4a54f294L75

File: content/millennium/r4/diagnostic/observation.md (hunk 12 above)
Comment (BS062721) [reply content/millennium/r4/diagnostic/observation.md]: https://github.com/cerner/fhir.cerner.com/pull/345/commits/3756e931418b52a36c0e6998015daa0d7adbb1c5#diff-d42a2575832583fdd3d20dff4a54f294R110

File: content/millennium/r4/diagnostic/observation.md (hunk 10 above)
Comment (BS062721) [reply content/millennium/r4/diagnostic/observation.md]: changed it to match subject: https://github.com/cerner/fhir.cerner.com/pull/345/commits/3756e931418b52a36c0e6998015daa0d7adbb1c5#diff-d42a2575832583fdd3d20dff4a54f294R69

File: content/millennium/r4/diagnostic/observation.md [hunk 13]
```diff
@@ -0,0 +1,124 @@
...
+    * [text](https://hl7.org/fhir/R4/observation-definitions.html#Observation.referenceRange.text){:target="_blank"}
+* [Member Observations (for limited use cases)](https://hl7.org/fhir/R4/observation-definitions.html#Observation.hasMember){:target="_blank"}
+* [Component (eg: systolic and diastolic for blood pressure)](https://hl7.org/fhir/R4/observation-definitions.html#Observation.component){:target="_blank"}
+
+## Terminology Bindings
+<%= terminology_table(:observation, :r4) %>
+
+## Extensions
+
+* valueAttachment: URL for this extension is defined as: `https://hl7.org/fhir/5.0/StructureDefinition/extension-Observation.valueAttachment`
```
Comment (MadhurRaj) [inline content/millennium/r4/diagnostic/observation.md]: should be http not https

Comment (BS062721) [reply content/millennium/r4/diagnostic/observation.md]: https://github.com/cerner/fhir.cerner.com/pull/345/commits/d95426d527f99a6e24c106ce3b53543266d49815#diff-d42a2575832583fdd3d20dff4a54f294R43

**Comment by BS062721** [comment]
Post deployment verification:
Screen Shot 2020-02-13 at 4 51 16 PM
Screen Shot 2020-02-13 at 4 51 21 PM
Screen Shot 2020-02-13 at 4 51 30 PM
Screen Shot 2020-02-13 at 4 51 34 PM

**Pull request merged**